scripts/o3de/ExportScripts/export_source_android.py

```diff
@@ -0,0 +1,252 @@
+
+    tools_build_path = eutil.handle_tools(ctx,
+                                          should_build_tools,
+                                          is_installer_sdk,
+                                          tool_config,
+                                          '',
+                                          tools_build_path,
+                                          default_base_path,
+                                          engine_centric,
+                                          logger)
+
+    eutil.build_and_bundle_assets(ctx, should_build_all_assets, tools_build_path, is_installer_sdk, 
+                                  tool_config, engine_centric, fail_on_asset_errors,
+                                  [ASSET_PLATFORM], validated_seedslist_paths, preprocessed_seedfile_paths, 
+                                  ctx.project_path/'AssetBundling', max_bundle_size,
+                                  logger)
+
+    android_project_config= android_support.get_android_config(project_path=None)
+
+    android_sdk_home = pathlib.Path(android_project_config.get_value(key=android_support.SETTINGS_SDK_ROOT.key))
+
+    parsed_android_generate_command = android_arg_parser.parse_args(['android-generate', '-p', str(ctx.project_name), '-B', str(target_android_project_path), '--asset-mode', asset_pack_mode])
+    android.generate_android_project(parsed_android_generate_command)
+
+    gradle_build_command = [target_android_project_path / f'gradlew{android_support.GRADLE_EXTENSION}', f'assemble{build_config.title()}']
+    has_errors = exp.process_command(gradle_build_command, cwd=target_android_project_path)
+    if has_errors:
+        raise exp.ExportProjectError("Gradle Build Failure.")
+
+    if deploy_to_device:
+        
+        activity_name = f'{ctx.project_name}Activity'
+        logger.info(f"Activity name set to {activity_name}")
+
+        if not org_name:
+            org_name = f'org.o3de.{ctx.project_name}'
+            logger.info(f"Organization name set to  {org_name}")
+
+        signing_config_file_path = target_android_project_path /'app/build.gradle'
+
+        try:
+            with open(signing_config_file_path,'r') as sf:
+                sf_text = sf.read()
+                signing_config_verified = ('signingConfigs' in sf_text and
+                                           'storeFile' in sf_text and
+                                           'storePassword' in sf_text and
+                                           'keyPassword' in sf_text and
+                                           'keyAlias' in sf_text)
+                if signing_config_verified:
+                    android_support.deloy_to_android_device(target_android_project_path,
+                                                build_config,
+                                                android_sdk_home,
+                                                org_name,
+                                                activity_name)
```

Comment:
1. `signing_config_path` is a pathlib object, which means you have access to the `[read_text](https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text)` method, so no need to read in the text manually. 
2. The `if signing_config_verified` line is indented to be inside the `with open(...` which is unnecessary, there was no need to hold onto the `sf` file handle at this point.
3. Doing an str `in` operation that many times on a potentially large text field is very inefficient. Using the `re` python module would help here

```suggestion
            build_gradle_content = signing_config_file_path.read_text('utf-8','ignore')
            match_signing_config = re.compile("(signingConfigs\s).*(storeFile\s).*(storePassword\s).*(keyPassword\s).*(keyAlias\s).*", re.DOTALL)
            signing_config_verified = match_signing_config.search(build_gradle_content) is not None
            if signing_config_verified:
                android_support.deloy_to_android_device(target_android_project_path,
                                            build_config,
                                            android_sdk_home,
                                            org_name,
                                            activity_name)
```